Enhance UI Feedback & Add Premium Question Submission Support

bencsci opened the pull request:
**Button & Status Indicators:** 
Added subtle animations (press-down effect) and loading spinners for a better user experience.

**Toast Notifications:** 
Implemented success / error toasts after every submission to give users immediate feedback.

**Premium Question Handling:**
Attempt submission via the standard /api/problems/${slug} endpoint.
On HTTP 404 – Not Found, retry using the api/premium/problems/${slug} endpoint.

This is actually my first pull request, so I’m not totally sure if I did everything right. Feel free to let me know!

Also, I used my own clientId and clientSecret for testing on my local machine.

bencsci committed:
UI updates for better feedback, support saving premium questions

bencsci committed:
Update background.js

cbkinase commented:
Hey there! Thanks for using and contributing to StructyHub. I ran out of premium access quite some time ago, so I really appreciate you pointing out this bug about premium questions - I had no idea!

For a first-time PR, I really like this. But I do have some comments and thoughts before I'm ready to merge it in:

1) (this is probably one of the most important, especially just as general advice) - but when you submit a PR, try to ensure your formatter settings match the project's. For example, there are over ~200 lines of the change that are just formatting (things like using 2 spaces instead of 4 in CSS files, or trailing commas). This obviously isn't a big deal *per se*, but as a reviewer it makes it harder to know where the actual important changes happened, and it can help ameliorate the problem of merge conflicts. In some ways, this is probably on me for not making formatting expectations clear, not having clear instructions for how to contribute to the project, etc, but I think it's useful general advice, and probably good practice even if the project is not explicit about it.

2) Personally, I'm not a huge fan of some of the styling changes. Actually, there is just one in particular I'd ask you to change -- I'll annotate the PR to show this line. I *do* think the toast notification is a nice add, however.

3) Nice resolution to the premium question handling. Another way we could do it, to reduce the number of API calls to api.structy.net, is check if the users' current URL has the string `/premium/` in it, and this way we could avoid doing two calls, but honestly, I'm quite happy with your current solution to it. I'm sure structy can handle one extra request :P

Overall, great job, especially for a first PR. I'm just about willing to accept it as-is if you can adjust those two main things (formatter settings and the one styling change I'll annotate). Thank you for your efforts in making StructyHub a true community project!

cbkinase commented on frontend/popup.css:
```diff
@@ -1,271 +1,346 @@
+
+#title-container {
+  display: flex;
+  background: rgb(6, 145, 121);
```
I'd recommend getting rid of the background here, as it makes the placement of "Report an Issue" look imbalanced. Alternatively, you could try to down-shift "Report an Issue" slightly to preserve balance, but IMO it looks better without title-container having the background anyway.

cbkinase commented on frontend/popup.html:
```diff
@@ -1,94 +1,143 @@
-<body>
+  <body>
+    <div id="title-container">
+      <h1 id="title">Structy Hub</h1>
```
StructyHub should be one word

cbkinase commented on frontend/popup.css:
```diff
@@ -1,271 +1,346 @@
+.ubuntu-light {
+  font-family: "Ubuntu", sans-serif;
+  font-weight: 300;
+  font-style: normal;
```
Am I missing something or do we never use this class anywhere?

cbkinase commented on frontend/popup.css:
```diff
@@ -1,271 +1,346 @@
+  padding-top: 10px;
+  padding-bottom: 10px;
+  border-radius: 5% 40px;
+  font-family: "Ubuntu", sans-serif;
```
Again, a personal preference thing, but I'm not a huge fan of how the 'lone Ubuntu' looks - I think we should either
a) be consistent with the usage of Ubuntu font, or
b) get rid of it entirely

and I'm probably leaning towards (b).

bencsci committed:
Format and Style Fixes

bencsci committed:
Merge branch 'main' of https://github.com/bencsci/StructyHub

bencsci committed:
Formatting Fixes

cbkinase commented on scripts/content_script.js:
```diff
@@ -178,7 +178,17 @@ function getProblemSlug() {
+  let path;
+
+  // Check if the problem is premium
+  if (url.includes("premium")) {
```
My one last final little nitpick is that I'd prefer this is `/premium/` (with the slashes) to specifically distinguish a premium problem from a problem that just-so-happens to have "premium" in the name. I actually don't think such a problem exists now, but it could save us potential headaches in the future if Alvin adds more problems :)

bencsci committed:
Update for premium checking logic

bencsci commented:
Thanks so much for your feedback! I really appreciate you taking the time to review this.

I've made the changes you suggested:

1. Fixed formatting to match the original (4 spaces for CSS, trailing commas for JS).
 
2. Reverted the style changes (title background, fonts, and name changes).

3. Implemented your suggestion of checking if the URL contains "premium" to save an API call.

Also, some insight on your comments:

.ubuntu-light: You’re right, this wasn’t used. I was experimenting with fonts and forgot to remove it.

Regarding using Ubuntu: I was aiming to match Structy’s title font, but I agree with you about consistency.

Overall, thank you for helping me out with my first pull request. I really appreciate your guidance! :)

cbkinase commented:
Excellent work @bencsci - thank you very much for your contribution! I'll work on getting this version submitted to the Chrome Web Store ASAP.
